### ISSUE TITLE
instant reeze at startup on mac 10.11.6

### ISSUE THREAD

**Issue by knobknows** [VCVRack/Rack#246]
OS (if applicable):10.11.6
Version (or "dev" if compiling from source):0.4.0

Hi, i m desperately trying to get Rack running, but whenever i start the app an empty black screen appears and the mac freezes instantly.
Tried several things, like disconnecting external soundcard, re-install etc. but still the same problem.

Im trying to get it running on a Mid 2010 Macbookpro with internal Intel GFX.

Hopefully, i can try it out soon!

**Comment by bontric** [comment]
Did you try to delete the autosave.vcv? 
This might be related to https://github.com/VCVRack/AudibleInstruments/issues/17

**Comment by knobknows** [comment]
where can i find that?

**Comment by bontric** [comment]
Should be in your Rack folder, but might be different on mac. 
This woild only work  if you had a previous version of rack installed.

**Comment by AndrewBelt** [comment]
`~/Documents/Rack/`

**Comment by knobknows** [comment]
thanks for the infos !
Nope, no autosave file in the Rack folder.
i should add that vcv rack could not be started properly by now. maybe thats why there is no autosafe file?

**Comment by bontric** [comment]
Yes, that's the reason. Now we know that the autosave file is not the problem.

Would you able to compile Rack from sources? This way you could help us out with some more detailed information by running a debugger. Otherwise it's more of a hit-or miss guessing what could be going on :wink:

**Comment by AndrewBelt** [comment]
@discohead Thanks for the advice, I didn't know that!

**Comment by knobknows** [comment]
I tried to build Rack on my machine locally, but didnt work.
The local dependencies have been built successfully, but whenever i try to build Rack itself i get this message:
c++ -Iinclude -Idep/include -Idep/lib/libzip/include -DVERSION=dev -MMD -O3 -march=nocona -ffast-math -Wall -g -DARCH_MAC -mmacosx-version-min=10.7 -DAPPLE -stdlib=libc++ -std=c++11 -stdlib=libc++ -c -o build/src/app.cpp.o src/app.cpp
In file included from src/app.cpp:1:
In file included from include/app.hpp:4:
include/widgets.hpp:5:10: fatal error: '../ext/nanovg/src/nanovg.h' file not found
#include "../ext/nanovg/src/nanovg.h"

**Comment by antoniotuzzi** [comment]
you didn't read the full instructions for the "FIRST" compile ;)

**Comment by knobknows** [comment]
?

**Comment by AndrewBelt** [comment]
Please read the Rack README

    git submodule update --init --recursive

**Comment by knobknows** [comment]
Alright,

now it worked, but still the same result. Whenever i start the app a black screen occurs, i guess as supposed, but the computer immediately freezes. No chance to switch apps, force quit etc. Only a forced shutdown is possible.
MacOs Console doesnt me anything relevant.

Whats wrong?

**Comment by antoniotuzzi** [comment]
did you try a **make clean**
and a new **make**
and new **make run**
?

**Comment by knobknows** [comment]
Yes, yes and yes!
and as already mentioned, no logs related to Rack.

Thanks for the help

**Comment by antoniotuzzi** [comment]
the obvious guess is an incompatibility openGL 
so libGLEW the culprit
maybe you should tweak some parameter using ./configure (in glew)

**Comment by bontric** [comment]
It would be really helpful if you could type:

```bash
make debug
```
The program will still crash but after the crash you will be dropped into a `gdb` shell. Just type `bt` in there and post the output. 

This will help a lot:wink:

**Comment by knobknows** [comment]
The OpenGL incompatibility might be the issue, thanks for the hint.
The debugging process doesnt work sine Rack causes the Computer to freeze, so nothings possible despite a brutal shutdown. 

Will try to tweak some gfx params.

thanks for the help!!

**Comment by knobknows** [comment]
OK, its definitely the gfx card (internal Intel HD Graphics 288mb VRAM).
Is there a way to run it with this gxf card?

**Comment by bontric** [comment]
Oh, sorry I didn't read properly and skipped the "Mac freezes instantly" part. You can try something else though

in a terminal, navigate to your rack folder. type:

```bash
./Rack >> rack.log
```

This writes the output from rack to a file (rack.log) which you can open after the mac freezes and you rebooted. Maybe this contains some useful information.

**Comment by knobknows** [comment]
Tried it, but the file is empty. 

The discreet gfx card is not usable, so i m preventing internal switching between the two gfx cards and therefore forcing to use the Intel HD Gfx. Which might be not compatible(?)
I guess thats the issue.

Thanks for the patience and the help

**Comment by knobknows** [comment]
Thanks, but this is definitely not the issue.
Its more the question whether the Intel HD Graphics card is compatible with Rack?
Mid 2010 Macbook Pro?
@AndrewBelt

**Comment by AndrewBelt** [comment]
Doesn't that have a NVIDIA GeForce 320M?
Anyway, it should definitely work on that Macbook. If the entire OS freezes, it is usually caused by the GPU: VRAM running out, or a driver kernel panic when a weird extension is loaded. The way I could go about solving this myself is to write a test program and start adding features to it until it does essentially the same thing as Rack. But since I don't have the computer in front of me, I can't solve this.

**Comment by knobknows** [comment]
Yes , but the GeForce Card is is not usable, only the built-in Intel HD GFX.
